### DISCUSSION TITLE
Background image not working

### DISCUSSION THREAD

**Discussion by tengkuzulfadli** [tailwindlabs/tailwindcss#5504] [Help] [answered]
Hi,

I already followed all the instructions that related to my issue and  I also installed the latest version, but the background image is still not working.

Can anyone spot any mistakes that I made, maybe from the project folders arrangement or wrongly used the src link on the config file?

Appreciate your assistance!!
![Screenshot (105)](https://user-images.githubusercontent.com/62935899/133374522-0274828e-f4d3-423c-854e-d2737052b7a3.png)
![Screenshot (106)](https://user-images.githubusercontent.com/62935899/133374527-2f2dc578-4afe-4c52-b2a1-1a6c5b234606.png)

**Comment by adamwathan** [comment]
Hey! To me it looks like an issue with your build setup + an issue with the URL. It looks like the site builds to `./build` but there is no `images` folder there (or `js` either) so you might have something wrong in your build configuration for whatever tooling you are using here that isn't copying those files.

The URL for the image likely shouldn't contain `src` as that's your development directory. Depends how you deploy your site, but if `/images` lives at the root of your site when deployed (like `example.com/images/origin_bg.png`) then you need to change the URL to `/images/origin_bg.png`.

To prove this isn't a bug in Tailwind, here's a play that has a custom background image utility configured:

https://play.tailwindcss.com/09br0HIh5c

**Reply by tengkuzulfadli** [reply]
hi @adamwathan ,

Thank you for your response. Yes I can see that I have an issue with my build set up. I already moved the image folder to Build folder and fix my config js file. But it's still not working. But my js file is now working well after I moved it to the build folder.
![Screenshot (108)](https://user-images.githubusercontent.com/62935899/133526693-2a109763-cf67-4a92-9063-4680b6eb7a07.png)
![Screenshot (110)](https://user-images.githubusercontent.com/62935899/133526702-3077deed-7f11-4b98-9be8-91cbcd0b8c24.png)


I'm wondering if is there any other ways to fix it?

**Reply by bheemisme** [reply]
Thanks for this, it worked for me.
I moved images folder to output folder and referenced image like this
```
"url('./images/image_name.png')" 
```

**Comment by simonswiss** [comment, marked as answer]
Hey!

You probably need to wrap the path inside the `url()` in quotes (use single/double/back ticks to avoid clashing with the sting quotes!):

```diff
- origins: "url(/image/origin_bg.png)"
+ origins: "url('/image/origin_bg.png')" 
```

If that doesn't work, make sure `/images` is in the root of your project. If it's not, you could think about the path you pass in `url()` in your CSS as relative to _where that CSS file is_, when the site is built. Even if you define it in your Tailwind config, that path will be placed inside the generated CSS, and the path needs to make sense _from there_

In your case, assuming your CSS file is generated in `build/css/yourfile.css`...

![image](https://user-images.githubusercontent.com/485747/133542746-5adfa0a8-764f-4212-af35-5f4db5e35065.png)


... to access the images folder, you need to first _go back up one level_ in your directory tree, with `../`.

And for that reason, the path in your `url()` should be:

```css
url('../images/origin_bg.png')
```

But I'm pretty sure your issue is the missing quotes around the path, inside the `url()`.

**Reply by EduardoSimon** [reply]
I had the exact same issue, thanks for the help! However, it's concerning that there is no error from the JIT compiler showing that there's a syntax error. In my case I am using postcss + esbuild to bundle my react lib with tailwind.

**Comment by michael-land** [comment]
Does JIT support background image yet?

**Comment by raycris** [comment]
### Tailwind config
I have the same issue but all see good but is not working

![Screenshot 2023-03-20 165933](https://user-images.githubusercontent.com/45718089/226465160-1fbfc364-8156-45fd-9c7e-e3ea3bf78bc4.jpg)

### Code
Here my code, I had to put a img label because with the Tailwind configuration isn't work, here you can see. As I mention before `bg-norway` is not working for me, I don't know why.
![Screenshot 2023-03-20 165156](https://user-images.githubusercontent.com/45718089/226465618-5e4eaac9-b9c1-4d4d-8247-c6231a159be1.jpg)

**Reply by wongjn** [reply]
You probably don't want the same path in the CSS definition as you do when you use `import`. This is because the `import` path is relative to the **source file**, whereas the CSS `url()` paths will be relative to **the page**. Consider trying something like `/assets/images/…` instead of `../assets/images/…`.

**Reply by raycris** [reply]
Thank you, for answer that fast but I try this before and didn't work for me. any other idea?

**Reply by wongjn** [reply]
Try to find the URL that your images are served at by directly navigating to them in the browser. You could consider reviewing your framework documentation to help you do this. For example, if you managed to navigate to an image via `http://localhost/public/assets/images/…`, then your values would be `/public/assets/images/…`. This is assuming that your framework automatically statically serves the images in its web server.

**Reply by raycris** [reply]
I managed to navigate using this url `http://127.0.0.1:5173/src/assets/images/norway.jpg` it's a localhost but still no working. I try to put `/src/assets/images/norway.jpg` in the tailwind config file as you mention but nothing.

**Comment by koachcode** [comment]
I manage to fix the issue, when i deployed my code to live, the background image does not show because it requires a relative path, I fixed it by adding **../** in from of the path and it worked both on localhost and on life.

## Folder Structure below 

![image](https://user-images.githubusercontent.com/58771507/234101032-cd8fb6b9-8aa4-461b-a2e3-6b6675a365cb.png)

## tailwind.config.js background Image extend 

![image](https://user-images.githubusercontent.com/58771507/234101208-7e0200e3-96a8-48be-b320-4e20e411508f.png)

## Usage in index.html

![image](https://user-images.githubusercontent.com/58771507/234101367-508dc19b-e43a-43dc-9829-772607a23a97.png)

**Comment by developervick** [comment]
`function Portfolio () {
    return(
        <React.Fragment>
            <div className="bg-[url('/img/hero-bg.jpg')] w-[800px] flex flex-col font-bold text-6xl justify-center items-center text-black">
                <h1 className="text-green-500">First react page</h1>
                <h2 className="text-zinc-700">My new Key board</h2>
            </div>
        </React.Fragment>        
        )
}` 

I am working in nextjs framework with tail wind css, In best of my knowldge i m following all the documentaion instruction

but my out put is like

![Screenshot (255)](https://user-images.githubusercontent.com/104270477/234905065-91ff40da-245d-44cb-b69f-1a10c4b0fda6.png)

**Comment by Canthar90** [comment]
In my case classes w-full h-full was not working changing h-full to h-screen resolved this problem
